File: src/main/java/com/imp/checklist/DefectType.java
package com.imp.checklist;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@AllArgsConstructor
@Getter
@Setter
@Document("defectType")
public class DefectType {
    @Id
    private String id;
    private String name;
}

File: src/main/kotlin/com/julia/imp/checklist/DefectType.kt
package com.julia.imp.checklist

import kotlinx.serialization.Contextual
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import org.bson.types.ObjectId

@Serializable
data class DefectType(
    @Contextual
    @SerialName("_id")
    val id: ObjectId,
    val name: String
)
